assets/src/edit-story/components/canvas/canvasProvider.js

```diff
@@ -70,6 +70,14 @@ function CanvasProvider( { children } ) {
 		if ( 'mousedown' === evt.type ) {
 			evt.persist();
 			setLastSelectionEvent( evt );
+
+			// Clear this selection event as soon as mouse is released
+			// `setTimeout` is currently required to not break functionality
+			// XXX: Is there a more elegant way to do this?
+			evt.target.ownerDocument.addEventListener(
+				'mouseup',
+				() => window.setTimeout( setLastSelectionEvent, 0, null ),
+			);
```

Comment:
@dvoytenko, if you have any suggestions as to how to clear this one better, please chime in. Without this, re-selecting a previously canvas-selected element from the layer panel will make the element stick to the mouse, which is of course a complete no-go. This "fixes" it, but it's not pretty.

Reply:
I think doing this in the `onMouseUp` in the `SelectionCanvas` is the way to go. It's already targeted for all different "selection" tasks - this is just one more.

Reply:
Good idea. But it doesn't work unfortunately. If I do what you suggest and then:

* select an element on the canvas,
* click-drag it to somewhere else,
* unselect it by clicking the background, and
* then re-select it from the layer panel,

It'll be stuck to the mouse after letting go of the mouse (it's just a simple click on the layer).

The above is the only way I've found to disable this weird behavior completely in all cases.

Reply:
Hmm. I'm a bit at loss how the "saved" event gets there then. If we cleared out the saved event when we clicked on the background, where would a new event come in the layer selection?

Reply:
That being said, I'm definitely ok if we do something like what you suggest. We should just then ensure we use `once` event listener and probably `capture` as well.

Reply:
I've `once`'d and `capture`'d it in the latest PR by the way.